upload_vehicle_data.py

```diff
@@ -0,0 +1,729 @@
+            logger.error(f'Failed to create directory {directory_path}: {e}')
+            raise
+
+
+def process_directory(
+    logger,
+    remote_user,
+    remote_ip,
+    remote_directory,
+    cloud_upload_directory,
+    config,
+    base_remote_directory,
+    bandwidth_mbps,
+):
+    """Process each directory."""
+    # Create the remote temporary directory
+    create_remote_temp_directory(
+        logger, remote_user, remote_ip, remote_directory
+    )
+
+    try:
+        # Find and copy the metadata.yaml file
+        metadata_path = find_metadata_file(
+            logger, remote_user, remote_ip, remote_directory
+        )
+        if metadata_path is None:
+            logger.error(
+                f'metadata.yaml file not found in {remote_directory}. '
+                f'Skipping.'
+            )
+        relative_metadata_path = os.path.relpath(
+            metadata_path, start=base_remote_directory
+        )
+
+        local_metadata_path = os.path.join(
+            cloud_upload_directory, relative_metadata_path
+        )
+
+        check_and_create_local_directory(
+            logger, os.path.dirname(local_metadata_path)
+        )
+
+        if not copy_metadata_file(
+            logger,
+            remote_user,
+            remote_ip,
+            metadata_path,
+            cloud_upload_directory,
+            base_remote_directory,
+        ):
+            logger.error(
+                f'Failed to copy metadata.yaml from '
+                f'{remote_directory}. Skipping.'
+            )
+
+        # Read the metadata.yaml file
+
+        metadata = read_metadata(logger, local_metadata_path)
+        expected_bags = metadata.get('rosbag2_bagfile_information', {}).get(
+            'relative_file_paths', None
+        )
+
+        # Get the list of rosbags from the remote machine
+        rosbag_list = get_remote_rosbags_list(
+            logger, remote_user, remote_ip, remote_directory
+        )
+
+        if len(rosbag_list) != len(expected_bags):
+            logger.error(
+                'The number of rosbags does not match the metadata. Skipping.'
+            )
+            return
+
+        rosbag_sizes = [
+            get_remote_file_size(logger, remote_user, remote_ip, rosbag)
+            for rosbag in rosbag_list
+        ]
+        total_size_gb = sum(rosbag_sizes)
+        estimated_time = get_estimated_upload_time(
+            total_size_gb, bandwidth_mbps, rosbag_sizes
+        )
+
+        if estimated_time < 1:
+            estimated_time_str = f'{estimated_time * 60:.2f} minutes'
+        else:
+            estimated_time_str = f'{estimated_time:.2f} hours'
+
+        logger.info(
+            f'Found {len(rosbag_list)} files to upload with total size '
+            f'{total_size_gb:.2f} GB from {remote_directory}.'
+        )
+        logger.info(
+            f'Estimated total time (including compression) '
+            f'is at least: {estimated_time_str}.'
+        )

```

Comment:
Please update this log to support the duration format HH:MM:SS. Specifically:

- When the duration includes hours, it should display as X hours M minutes S seconds.
- If the duration is less than an hour, it should display as M minutes S seconds.
- If the duration is less than a minute, it should display as S seconds